designdocs/provider_contexts.md

```diff
@@ -0,0 +1,379 @@
+Additionally, to limit the impact on the network, the Bar and Baz Applications are able to share capacity and perform batching together.
+This leads to the conclusion that Provider-level delegations are better than introducing new Application-level delegations.
+While user permissions are primarily write-based at this time, it is critical that users understand the delegation and their data is shared at a company level.
+
+Finally, this structure does not prevent the setup of a 1:1 relationship between Application and Provider.
+If a company wishes, they can create more than one Provider for the purposes of separation.
+This may be desirable if there is a subsidiary company or other complex structures.
+
+#### User Trust in Application Presentation
+
+Users need to know what application they are logging into and trust that the displayed information is not a phishing attack.
+Users have a reasonable expectation that the chain and wallet provide a level of protection against phishing attacks from Providers and Applications.
+
+#### Easy Wallet Integration of Data
+
+Wallets displaying login request information need to be able to easily obtain and verify the information they want to show the user.
+
+## Proposal
+
+Frequency should provide a safeguarded system of setting and updating Provider and Provider Application Contexts for wallets to display to users.
+
+- [Storage and Structure of the Data](#data)
+- [Changes and additions in extrinsics](#extrinsics)
+- [Storage Migration](#migration)
+- [Mainnet Approval Flow](#governance)
+- [Example of Wallet Usage via SIWF](#siwf)
+- [Provider Dashboard Steps](#dashboard)
+
+### **Storage and Structure of the Data** <a id='data'></a>
+
+Currently, the `ProviderRegistryEntry` has only a `name` property, and no way to update said name.
+
+Applications have two different pieces of default and (potentially) internationalized data:
+
+- name
+- logo/image
+
+Limits:
+
+1. Provide a space to internationalize up to 150 localizations (Windows and macOS each have < 150).
+2. Image MUST BE encoded (`png`)[https://www.w3.org/TR/png-3/] that is `250x100` (support for future sizes and light/dark should be considered).
+3. Localization keys should follow BCP 47 format like en-US, fr-FR.
+
+The data structure must support both of these and internationalization.
+
+Proposed are the following changes:
+
+1. Update `ProviderRegistryEntry`.
+
+    Each registry entry supports:
+    - A default name and logo (hash of the logo image).
+    - Localized names and logos (up to 150 locales).
+    - PNG logo support (250x100, up to 128 KiB). Note that the logo is not embedded in the proposal but rather a hash of the logo image.
+    - BCP 47-compliant language codes (e.g., en-US, fr-FR).
+2. Updated `ProviderRegistryEntry` struct have the following properties:
+
+    ```rust
+    type LogoHash = [u8; 32]; // blake2_256 hash of the logo
```

Comment:
What if LogoHash was instead the CID? Stay with me here, as I'm not sure this works:

1. LogoHash is the CID. This way when submitted to the chain for approval, there is a simple way for the approval process to check the image.
2. (Not sure if this works) Submission of the image to be added to `ApprovedLogos` would need to check that the image matches the given CID. (I think this would require making some CID assumptions or documenting the defaults?)

Reply:
This is doable and we already have some basic CID validation in `messages` for IPFS, my only question here is that is it ok to?

1. Provider may pin a logo and get CID
2. Proposes application with CID as logo hash
3. Upon approval submits a logo with cid but it might be a different file than originally pinned to bypass some governance restrictions 

If we get a CID is it possible to have some offchain worker that creates an offchain index and pull the image from CID and compute a hash of actual image for verification but i guess that might be an overkill

Reply:
As long as we are able to validate the CID on chain, then the file must be the same. As the CID has a sha2-256 hash "inside" of it. Ok not quite inside of it and this is where the question sits. Since it isn't just the bytes being hashed: https://docs.ipfs.tech/concepts/content-addressing/#cids-are-not-file-hashes

Reply:
Re imagined LogoHash -> LogoCID,  atleast on the messages side we don't do a full CIDv1 validation, here we can recompute on file upload ( not sure if that would be possible) .

Following same rules around CIDv1 as in our message design.